doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -552,90 +533,72 @@ the conversion in hypotheses :n:`{+ @ident}`.
+ The :math:`\delta` rule only applies to transparent constants (see :ref:` controlling-the-reduction-strategies`
  on transparency and opacity).
 
-.. tacn:: cbn
-          simpl
-   :name: cbn; simpl
+.. tacn:: simpl {? @delta_flag } {? {| @reference_occs | @pattern_occs } } {? @occurrences }
 
-   These tactics apply to any goal. They try to reduce a term to
-   something still readable instead of fully normalizing it. They perform
+   Reduces a term to
+   something still readable instead of fully normalizing it. It performs
    a sort of strong normalization with two key differences:
 
-   + They unfold a constant if and only if it leads to a :math:`\iota`-reduction,
+   + It unfolds constants only if they lead to an :math:`\iota`-reduction,
      i.e. reducing a match or unfolding a fixpoint.
-   + While reducing a constant unfolding to (co)fixpoints, the tactics
-     use the name of the constant the (co)fixpoint comes from instead of
+   + When reducing a constant unfolding to (co)fixpoints, the tactic
+     uses the name of the constant the (co)fixpoint comes from instead of
      the (co)fixpoint definition in recursive calls.
 
-   The :tacn:`cbn` tactic was intended to be a more principled, faster and more
-   predictable replacement for :tacn:`simpl`.
-
-   The :tacn:`cbn` tactic accepts the same flags as :tacn:`cbv` and
-   :tacn:`lazy`. The behavior of both :tacn:`simpl` and :tacn:`cbn`
-   can be tuned using the :cmd:`Arguments` command.
+   :n:`@occurrences`
+     Permits selecting whether to reduce the conclusion and/or one or more
+     hypotheses.  The `at` option allows selection of specific occurrences
+     matching :n:`@pattern_occs`.
 
-   .. todo add "See <subsection about controlling the behavior of reduction strategies>"
-      to TBA section
+   Only transparent constants whose name can be reused in
+   recursive calls may be unfolded by :tacn:`simpl`. For instance, a
+   constant :g:`plus' := plus` may be unfolded and reused in
+   recursive calls, but a constant such as :g:`succ := plus (S O)` is
+   never unfolded.
 
-   Notice that only transparent constants whose name can be reused in the
-   recursive calls are possibly unfolded by :tacn:`simpl`. For instance a
-   constant defined by :g:`plus' := plus` is possibly unfolded and reused in
-   the recursive calls, but a constant such as :g:`succ := plus (S O)` is
-   never unfolded. This is the main difference between :tacn:`simpl` and :tacn:`cbn`.
-   The tactic :tacn:`cbn` reduces whenever it will be able to reuse it or not:
-   :g:`succ t` is reduced to :g:`S t`.
+   :n:`{| @reference_occs | @pattern_occs }` can limit the application of :tacn:`simpl`
+   to:
 
-.. tacv:: cbn [ {+ @qualid} ]
-          cbn - [ {+ @qualid} ]
+   - the applicative subterms whose head occurrence is the unfoldable
+     constant :n:`@qualid` or is the constant used
+     in the notation :n:`@string`
```

Comment:
`@qualid` and `@string` come from the definition of `@reference`. The text should make that explicit (e.g., "see the definition of `@reference`").

Reply:
> Also, what happens if there are multiple constants used in the notation `@string`?

An exercise for the reader, unless this is important to cover and you can give a simple explanation.

Reply:
The simple explanation is that it's the head constant.  (I don't recall whether heading the term with a match, forall, fun, etc, results in an error, a noop, or results in picking the head application node under the match/forall/fun)

Reply:
I believe the original text called it the "discriminating constant" or something?  (I think it's important to indicate that when multiple constants are used in the notation, Coq has a simple algorithm that selects which one to use.  Spelling out that algorithm would be nice, but is less important.)

Reply:
Here's the original.  There is no definition of "discriminating symbol".  

```
      If :n:`@string` denotes the discriminating symbol of a notation
      (e.g. "+") or an expression defining a notation (e.g. `"_ +
      _"`), and this notation denotes an application whose head symbol
      is an unfoldable constant, then the tactic unfolds it.
```

> I think it's important to indicate that when multiple constants are used in the notation, Coq has a simple algorithm that selects which one to use. Spelling out that algorithm would be nice, but is less important.

It's a tease to say there is a simple algorithm without giving any description.

It's hard for me to write anything more here because I don't understand most of the details.